crash: missing error check when loading config snapshot

There has been a change which is causing an error to appear when loading a config snapshot. The error is not checked and the nil snapshot is passed to the configloader causing a panic.

From https://github.com/hashicorp/terraform-provider-aws/issues/17770

```
    resource_aws_db_instance_test.go:483: Step 2/2 error: Error running destroy: exit status 2
        panic: runtime error: invalid memory address or nil pointer dereference
        [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb173ba]
        
        goroutine 11 [running]:
        github.com/hashicorp/terraform/configs/configload.(*Loader).ImportSourcesFromSnapshot(0xc00009d5e0, 0x0)
        	/home/circleci/project/project/configs/configload/loader.go:143 +0x3a
        github.com/hashicorp/terraform/backend/local.(*Local).context(0xc0005cbc70, 0xc0001a8b60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        	/home/circleci/project/project/backend/local/backend_local.go:108 +0x8d2
        github.com/hashicorp/terraform/backend/local.(*Local).opApply(0xc0005cbc70, 0x2ca1aa0, 0xc0002529c0, 0x2ca1aa0, 0xc000252a00, 0xc0001a8b60, 0xc000252980)
        	/home/circleci/project/project/backend/local/backend_apply.go:51 +0x2bf
        github.com/hashicorp/terraform/backend/local.(*Local).Operation.func1(0xc00037da50, 0xc00037da60, 0xc00037da70, 0xc0005cbc70, 0xc00037da40, 0x2ca1aa0, 0xc0002529c0, 0x2ca1aa0, 0xc000252a00, 0xc0001a8b60, ...)
        	/home/circleci/project/project/backend/local/backend.go:345 +0xe8
        created by github.com/hashicorp/terraform/backend/local.(*Local).Operation
        	/home/circleci/project/project/backend/local/backend.go:339 +0x36d
--- FAIL: TestAccAWSDBInstance_IsAlreadyBeingDeleted (410.17s)
```

It's unclear what the error is in this case, but the panic needs to be fixed regardless.

--- a/backend/local/backend_local.go
+++ b/backend/local/backend_local.go
@@ -95,6 +95,10 @@ func (b *Local) context(op *backend.Operation) (*terraform.Context, *configload.
 		}
 		log.Printf("[TRACE] backend/local: building context from plan file")
 		tfCtx, configSnap, ctxDiags = b.contextFromPlanFile(op.PlanFile, opts, stateMeta)
+		if ctxDiags.HasErrors() {
+			return nil, nil, nil, ctxDiags
+		}
+
 		// Write sources into the cache of the main loader so that they are
 		// available if we need to generate diagnostic message snippets.
 		op.ConfigLoader.ImportSourcesFromSnapshot(configSnap)

